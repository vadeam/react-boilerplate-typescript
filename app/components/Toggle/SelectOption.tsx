/**
 * SelectOption
 */

import * as React from 'react'
import { injectIntl, MessageDescriptor, WrappedComponentProps } from 'react-intl'

type SelectOptionProps = { key: string; value: string | number; message?: MessageDescriptor } & WrappedComponentProps

const SelectOption = ({ value, key, message, intl }: SelectOptionProps) => (
  <option value={value} key={key}>
    {message ? intl.formatMessage(message) : value}
  </option>
)

export default injectIntl(SelectOption)
