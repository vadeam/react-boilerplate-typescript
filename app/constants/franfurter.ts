export const DEFAULT_BASE_CURRENCY = 'EUR'
