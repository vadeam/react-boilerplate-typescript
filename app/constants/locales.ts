export const LOCALE_DE = 'de'
export const LOCALE_EN = 'en'
export const DEFAULT_LOCALE = LOCALE_EN

export const appLocales: Locales[] = [LOCALE_EN, LOCALE_DE]
export const CUSTOM_LOCALE_STORAGE_KEY = 'custom-locale'
