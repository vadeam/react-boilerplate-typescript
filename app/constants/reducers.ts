export const REDUCER_KEY_APP = 'app'
export const REDUCER_KEY_LANGUAGE = 'language'
export const REDUCER_KEY_ROUTER = 'router'
export const REDUCER_KEY_HOME = 'home'
// for testing purposes
export const REDUCER_KEY_TEST = 'test'
