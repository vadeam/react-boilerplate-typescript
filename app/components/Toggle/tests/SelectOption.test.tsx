import { render } from '@testing-library/react'
import React from 'react'
import { IntlProvider, defineMessages } from 'react-intl'

import SelectOption from '../SelectOption'
import { LOCALE_EN, LOCALE_DE } from '../../../constants/locales'

describe('<SelectOption />', () => {
  it('should render default language messages', () => {
    const message = defineMessages({
      enMessage: {
        id: `containers.LocaleToggle.${LOCALE_EN}`,
        defaultMessage: 'someContent',
      },
    })
    const { container } = render(
      <IntlProvider locale={LOCALE_EN}>
        <SelectOption key="key" value={LOCALE_EN} message={message.enMessage} />
      </IntlProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should display `value`(two letter language code) when `message` is absent', () => {
    const { queryByText } = render(
      <IntlProvider locale={LOCALE_DE}>
        {/* @ts-ignore */}
        <SelectOption value={LOCALE_DE} />
      </IntlProvider>,
    )
    expect(queryByText(LOCALE_DE)).toBeInTheDocument()
  })
})
