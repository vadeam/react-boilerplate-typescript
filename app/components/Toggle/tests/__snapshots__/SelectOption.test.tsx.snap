// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectOption /> should render default language messages 1`] = `
<option
  value="en"
>
  someContent
</option>
`;
